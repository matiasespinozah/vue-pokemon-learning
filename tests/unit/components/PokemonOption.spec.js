import PokemonOption from '@/components/PokemonOption'
import { shallowMount } from '@vue/test-utils';
import { mockPokemons } from '../mocks/pokemons.mock';

describe('PokemonOption component', () => {
  let wrapper = shallowMount(PokemonOption, {
    props: {
      pokemons: mockPokemons
    }
  })

  beforeEach(() => {
    wrapper = shallowMount(PokemonOption, {
      props: {
        pokemons: mockPokemons
      }
    })
  })

  test('debe de hacer match con snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('debe de mostrar las 4 opciones correctamentes', () => {
    const items = wrapper.findAll('li')
    expect(items.length).toBe(4)

    expect(items[0].text()).toBe('bulbasaur')
    expect(items[1].text()).toBe('ivysaur')
    expect(items[2].text()).toBe('venusaur')
    expect(items[3].text()).toBe('charmander')
  })

  test('debe de emitir "selection" con sus respectivos parametros al hacer click', () => {
    const [li1, li2, li3, li4] = wrapper.findAll('li')
    li1.trigger('click')
    li2.trigger('click')
    li3.trigger('click')
    li4.trigger('click')

    expect(wrapper.emitted('selection').length).toBe(4)
    expect(wrapper.emitted('selection')[0]).toEqual([1])
    expect(wrapper.emitted('selection')[1]).toEqual([2])
    expect(wrapper.emitted('selection')[2]).toEqual([3])
    expect(wrapper.emitted('selection')[3]).toEqual([4])
  })
  
})