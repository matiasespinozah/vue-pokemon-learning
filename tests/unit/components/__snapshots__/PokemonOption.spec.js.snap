// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOption component debe de hacer match con snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>bulbasaur</li>
    <li>ivysaur</li>
    <li>venusaur</li>
    <li>charmander</li>
  </ul>
</div>
`;
